goaviatrix/smart_group.go

```diff
@@ -35,55 +38,135 @@ type SmartGroup struct {
 	Selector SmartGroupSelector
 }
 
-func smartGroupFilterToMap(filter *SmartGroupMatchExpression) map[string]string {
-	filterMap := make(map[string]string)
+type SmartGroupResult struct {
+	UUID     string              `json:"uuid"`
+	Name     string              `json:"name"`
+	Selector SmartGroupAnyResult `json:"selector"`
+}
 
-	if len(filter.Type) > 0 {
-		filterMap["type"] = filter.Type
-	}
-	if len(filter.CIDR) > 0 {
-		filterMap["cidr"] = filter.CIDR
-	}
-	if len(filter.FQDN) > 0 {
-		filterMap["fqdn"] = filter.FQDN
-	}
-	if len(filter.Site) > 0 {
-		filterMap["site"] = filter.Site
-	}
-	if len(filter.ResId) > 0 {
-		filterMap["res_id"] = filter.ResId
-	}
-	if len(filter.AccountId) > 0 {
-		filterMap["account_id"] = filter.AccountId
-	}
-	if len(filter.AccountName) > 0 {
-		filterMap["account_name"] = filter.AccountName
-	}
-	if len(filter.Name) > 0 {
-		filterMap["name"] = filter.Name
-	}
-	if len(filter.Region) > 0 {
-		filterMap["region"] = filter.Region
-	}
-	if len(filter.Zone) > 0 {
-		filterMap["zone"] = filter.Zone
-	}
-	if len(filter.K8sClusterId) > 0 {
-		filterMap["k8s_cluster_id"] = filter.K8sClusterId
-	}
-	if len(filter.K8sNamespace) > 0 {
-		filterMap["k8s_namespace"] = filter.K8sNamespace
-	}
-	if len(filter.K8sService) > 0 {
-		filterMap["k8s_service"] = filter.K8sService
+type SmartGroupAnyResult struct {
+	Any []SmartGroupMatchExpressionResult `json:"any"`
+}
+
+type SmartGroupMatchExpressionResult struct {
+	All map[string]interface{} `json:"all"`
+}
+
+type SmartGroupResp struct {
+	UUID        string             `json:"uuid"`
+	SmartGroups []SmartGroupResult `json:"app_domains"`
+}
+
+const (
+	TagsPrefix      = "tags"
+	ExtArgsPrefix   = "ext_args"
+	CidrKey         = "cidr"
+	FqdnKey         = "fqdn"
+	TypeKey         = "type"
+	SiteKey         = "site"
+	ResIdKey        = "res_id"
+	AccountIdKey    = "account_id"
+	AccountNameKey  = "account_name"
+	NameKey         = "name"
+	RegionKey       = "region"
+	ZoneKey         = "zone"
+	K8sClusterIdKey = "k8s_cluster_id"
+	K8sNamespaceKey = "k8s_namespace"
+	K8sServiceKey   = "ka8s_service"
+	K8sPodNameKey   = "ka8s_pod"
```

Comment:
It seems like a typo sneaked in:
```suggestion
	K8sServiceKey   = "k8s_service"
	K8sPodNameKey   = "k8s_pod"
```

Reply:
Weird, I thought this was fixed. 
Seems like it went into master.